### PR TITLE
Add missing POSIX Error code for WASI OS

### PR DESCRIPTION
**Pull request by yonihemi**

### CODE REVIEW

File: stdlib/public/Platform/POSIXError.swift [hunk 1]
```diff
@@ -556,6 +556,168 @@ public enum POSIXErrorCode : Int32 {
   case EHWPOISON       = 133
 }
 
+#elseif os(WASI)
+
+/// Enumeration describing POSIX error codes.
+public enum POSIXErrorCode : Int32 {
```
Comment (MaxDesiatov) [inline stdlib/public/Platform/POSIXError.swift:562]: Since WASI is not POSIX, shouldn't this be named `WASIErrorCode`?

Comment (kateinoigakukun) [reply stdlib/public/Platform/POSIXError.swift:562]: WASI doesn't provide POSIX API, but wasi-libc is clearly POSIX compatible according to their readme.
> WASI Libc is sufficiently stable and usable for many purposes, as most of the POSIX-compatible APIs are stable, though it is continuing to evolve to better align with wasm and WASI.

https://github.com/WebAssembly/wasi-libc

Comment (MaxDesiatov) [reply stdlib/public/Platform/POSIXError.swift:562]: Cool, thanks for the clarification.

**Review by MaxDesiatov — APPROVED** [review]
Seems legit

**Pull request merged**